elements/lisk-chain/src/data_access/data_access.ts

```diff
@@ -343,7 +343,7 @@ export class DataAccess {
 		const transactions =
 			// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
 			blockJSON.transactions?.map(transaction =>
-				this._transactionAdapter.fromJSON(transaction),
+				this._transactionAdapter.decode(transaction),
```

Comment:
Maybe rename `transaction` to `transactionBuffer` so as it to be easy to tell it's a binary buffer?

Reply:
this piece of code is being refactored by shu in chain PR, so to avoid conflicting changes, I kept like this.